test/canny.cpp

```diff
@@ -95,31 +96,31 @@ void cannyImageOtsuTest(string pTestFile, bool isColor)
 
     for (size_t testId=0; testId<testCount; ++testId) {
 
-        af_array inArray  = 0;
-        af_array outArray = 0;
-        af_array goldArray= 0;
-        dim_t nElems   = 0;
-
         inFiles[testId].insert(0,string(TEST_DIR "/CannyEdgeDetector/"));
         outFiles[testId].insert(0,string(TEST_DIR "/CannyEdgeDetector/"));
 
-        ASSERT_EQ(AF_SUCCESS, af_load_image(&inArray, inFiles[testId].c_str(), isColor));
-        ASSERT_EQ(AF_SUCCESS, af_load_image_native(&goldArray, outFiles[testId].c_str()));
-        ASSERT_EQ(AF_SUCCESS, af_get_elements(&nElems, goldArray));
+        try {
```

Comment:
It would have been good if you didn't refactored this file. The tests are failing and I don't know the cause. I noticed that the original test is using af_load_image_native. and your example isn't. Is that intentional?

Reply:
I don't see how this is related the `clEqueueReadBuffer` failure on OSX. I have verified the inputs manually.

May be these are stemming from using C++ API. I will refactor this to C and check what happens.

Reply:
I want to have commits that serve one purpose. In this PR you are refactoring canny and modifying mean shift. Having separate commits allows us to pinpoint issues if there are errors in the future and it gives you an understanding why a certain change was made. For example, I don't know why you changed this test. If I looked at the commit message I would see that you modified the mean shift filter which will not give me information about this change.